src/utils/dotnetUtils.ts

```diff
@@ -6,112 +6,114 @@
         await validateDotnetInstalled();
+
         const listOutput: string = await cpUtils.executeCommand(undefined, undefined, 'dotnet', 'new', '--list');
-        if (!listOutput.includes(funcProjectId) || !listOutput.includes('HttpTrigger')) {
-            const installTemplates: MessageItem = { title: localize('installTemplates', 'Install Templates') };
-            await ui.showWarningMessage(localize('noDotnetFuncTemplates', 'You must have the Azure Functions templates installed for the .NET CLI.'), { modal: true }, installTemplates, DialogResponses.cancel);
-            await installDotnetTemplates(ui, outputChannel);
+        const templatesInstalled: boolean = listOutput.includes(funcProjectId) && listOutput.includes('HttpTrigger');
+
+        const templatesVersion: string | undefined = templatesInstalled ? ext.context.globalState.get(templatesKey) : undefined;
+        const cliFeedJson: cliFeedJsonResponse | undefined = await tryGetCliFeedJson();
+        if (!cliFeedJson) {
+            if (templatesInstalled) {
+                return;
+            } else {
+                throw new Error(localize('retryInternet', 'There was an error in retrieving the templates.  Recheck your internet connection and try again.'));
+            }
         }
-    }
-
-    export async function uninstallDotnetTemplates(outputChannel: OutputChannel): Promise<void> {
-        await validateDotnetInstalled();
 
-        const tempFolder: string = os.tmpdir();
+        let runtime: ProjectRuntime | undefined = await functionRuntimeUtils.tryGetLocalRuntimeVersion();
+        if (runtime === undefined) {
+            const picks: IAzureQuickPickItem<ProjectRuntime>[] = [
+                { label: ProjectRuntime.beta, description: '(.NET Core)', data: ProjectRuntime.beta },
+                { label: ProjectRuntime.one, description: '(.NET Framework)', data: ProjectRuntime.one }
+            ];
+            const options: QuickPickOptions = { placeHolder: localize('pickTemplateVersion', 'Select the template version to install') };
+            runtime = (await ext.ui.showQuickPick(picks, options)).data;
+        }
 
-        outputChannel.show();
-        outputChannel.appendLine(localize('uninstallFuncProject', 'Uninstalling "{0}"', projectPackageId));
-        await cpUtils.executeCommand(undefined, tempFolder, 'dotnet', 'new', '--uninstall', projectPackageId);
-        outputChannel.appendLine(localize('uninstallFuncs', 'Uninstalling "{0}"', functionsPackageId));
-        await cpUtils.executeCommand(undefined, tempFolder, 'dotnet', 'new', '--uninstall', functionsPackageId);
-        outputChannel.appendLine(localize('finishedUninstallingTemplates', 'Finished uninstalling Azure Functions templates.'));
-    }
+        const latestRelease: string = await tryGetTemplateVersionSetting(actionContext, cliFeedJson, runtime) || cliFeedJson.tags[getFeedRuntime(runtime)].release;
```

Comment:
Anything need to be guarded in this expression?

Reply:
They've promised that the cli feed will be backwards compatible, so I'm not too worried. I added some additional telemetry/error handling just in case, though.